### ISSUE TITLE
How to turn on inlay hints?

### ISSUE THREAD

**Issue by jestarray** [kitamstudios/rust-analyzer.vs#35]
![inlay hint](https://user-images.githubusercontent.com/34615798/108774134-2118c800-7514-11eb-8813-9ae9049493c2.png)

as you can see, the calls to `test` in `main()` show light text of the function parameter names in vscode. I don't know how to get this working with visual studio but it's built into rust analyzer. I think C/C++ has this feature too

**Comment by parthopdas** [comment]
My apologies, I might have mislead you on the YT comments. I misunderstood what you meant by inlay hints. This is indeed an useful feature.

However it seems to be missing from VS2022. 

Here is C++
![image](https://github.com/kitamstudios/rust-analyzer.vs/assets/2140460/41fbd764-5c27-4304-84ca-feb01b5efb7e)

Here is C#
![image](https://github.com/kitamstudios/rust-analyzer.vs/assets/2140460/7c3ba80a-a715-40e8-ab04-9104aeeb09aa)

**Comment by parthopdas** [comment]
okay inlay hints are present for both C++ and C#. Need to press Alt+F1. Ref https://stackoverflow.com/questions/71788959/visual-studio-community-2022-inline-hints-shortcut

**Comment by jestarray** [comment]
Yeah, I recall this not working with rust in vstudio but will need to double check

**Comment by parthopdas** [comment]
okay investigate this.

1. inlay hints are supported by LSP https://github.com/microsoft/vscode/issues/113276
2. it is possible for LSP clients packages in VS2022 to invoke the LSP in addition to the usual comms between VS2022 and LSP like rust-analyzer see https://learn.microsoft.com/en-us/visualstudio/extensibility/adding-an-lsp-extension?view=vs-2022#send-custom-messages
3. 2 is implemented in r-a.vs in LanguageClient.cs

the fix would be to do the same thing C# does - on Alt+F1, r-a.vs invokes the LSP using step 3, get back the inlay hints and display it using editor adornments. 

it is also is a good idea to add this plumbing for further enhancements that VS2022 does not provide out of the box but LSPs do. e.g. unit test run / debug code lenses.

the fix would take a couple of days. i would welcome contribution from the community. 

alternatively i can get to when when i have some free time.

**Comment by tausifcreates** [comment]
any updates on this?

**Comment by parthopdas** [comment]
None unfortunately. January is the earliest I'll have free time and it should be implemented then.

**Comment by rasimv** [comment]
How to disable inlay hints?
![rustanq](https://github.com/kitamstudios/rust-analyzer.vs/assets/5728975/539b172e-65c4-4196-b1a9-a59838d79b47)
